### PR TITLE
feat(AllowedFlags): Adds allowed_flags option

### PR DESCRIPTION
**Pull request by sezaru**
### Contributor checklist
- [x] My commit messages follow the [Conventional Commit Message Format](https://gist.github.com/stephenparish/9941e89d80e2bc58a153#format-of-the-commit-message)
      For example: `fix: Multiply by appropriate coefficient`, or
      `feat(Calculator): Correctly preserve history`
      Any explanation or long form information in your commit message should be
      in a separate paragraph, separated by a blank line from the primary message
- [ ] Bug fixes include regression tests
- [x] Features include unit/acceptance tests

### CODE REVIEW

**Comment by sezaru** [comment]
This PR is related to the issue https://github.com/zachdaniel/git_ops/issues/57

File: lib/mix/tasks/git_ops.release.ex [hunk 1]
```diff
@@ -275,7 +279,26 @@ defmodule Mix.Tasks.GitOps.Release do
...
   end
 
-  defp commits_with_type(config_types, commits, text, log?) do
+  defp commits_with_allowed_tags(commits, :any), do: commits
+
+  defp commits_with_allowed_tags(commits, allowed_tags) do
+    case Enum.find(commits, fn %{type: type} -> type == "TAGS" end) do
+      nil ->
+        # If the commits don't have a tag, we allow it to go through by default
+        commits
```
Comment (zachdaniel) [inline lib/mix/tasks/git_ops.release.ex]: we should add a config, i.e `allow_untagged?: true/false`

File: lib/mix/tasks/git_ops.release.ex [hunk 2]
```diff
@@ -90,6 +90,7 @@ defmodule Mix.Tasks.GitOps.Release do
     prefix = Config.prefix()
 
     config_types = Config.types()
+    allowed_tags = Config.allowed_tags() |> IO.inspect(label: "allowed_tags")
```
Comment (zachdaniel) [inline lib/mix/tasks/git_ops.release.ex]:
```suggestion
    allowed_tags = Config.allowed_tags()
```

**Review by zachdaniel — COMMENTED** [review]
LGTM, but I think we should add the config I mentioned.

File: lib/mix/tasks/git_ops.release.ex (hunk 1 above)
Comment (sezaru) [reply lib/mix/tasks/git_ops.release.ex]: Any suggestion on where that should be? is it OK if it is in the "root" level of the :git_ops config?

`config :git_ops, allows_untagged?: false`

**Comment by sezaru** [comment]
Maybe like this?

```elixir
config :git_ops,
  tags: [
    allow_untagged?: true
    allowed: []
  ]
```

**Review by zachdaniel — APPROVED** [review]

**Pull request merged**